.github/workflows/main.yml

```diff
@@ -35,30 +61,70 @@ jobs:
           version: "23.4"
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: snippets/rust -> snippets/rust/target
+
+      - name: temporarily get sdk
+        uses: actions/checkout@v3
+        with:
+          repository: breez/breez-sdk
+          ref: ${{ needs.setup.outputs.sdk-ref }}
+          path: breez-sdk
+
+      - id: rev-parse
+        name: get proper rev
+        working-directory: breez-sdk
+        run: |
+          rev=$(git rev-parse HEAD)
+          echo "$rev"
+          echo "rev=$rev" >> $GITHUB_OUTPUT   
+
+      - name: set sdk version
+        working-directory: snippets/rust
+        run: |
+          sed -i.bak -e 's/breez-sdk-core.*/breez-sdk-core = { git = "https:\/\/github.com\/breez\/breez-sdk", rev = "${{ steps.rev-parse.outputs.rev }}" }/' Cargo.toml
```

Comment:
The search-replace looks brittle.

How about this instead:

```bash
cargo add --git https://github.com/breez/breez-sdk.git breez-sdk-core --rev "${{ steps.rev-parse.outputs.rev }}"
```

It will update the `breez-sdk-core` dependency in `Cargo.toml` with that specific commit.

Reply:
Done